pkg/agent/apiserver/handlers/podinterface/handler_test.go

```diff
@@ -0,0 +1,237 @@
+	"192.168.0.2",
+}
+
+var macStrs = []string{
+	"00:00:00:00:00:00",
+	"00:00:00:00:00:01",
+	"00:00:00:00:00:02",
+}
+
+var macs = []net.HardwareAddr{
+	parseMAC(macStrs[0]),
+	parseMAC(macStrs[1]),
+	parseMAC(macStrs[2]),
+}
+
+var responses = []Response{
+	{
+		PodName:       "pod0",
+		InterfaceName: "interface0",
+		IP:            ipStrs[0],
+		MAC:           macStrs[0],
+		PortUUID:      "portuuid0",
+		OFPort:        0,
+		ContainerID:   "containerid0",
+		PodNamespace:  "namespace0",
+	},
+	{
+		PodName:       "pod1",
+		InterfaceName: "interface1",
+		IP:            ipStrs[1],
+		MAC:           macStrs[1],
+		PortUUID:      "portuuid1",
+		OFPort:        1,
+		ContainerID:   "containerid1",
+		PodNamespace:  "namespace1",
+	},
+	{
+		PodName:       "pod2",
+		InterfaceName: "interface2",
+		IP:            ipStrs[2],
+		MAC:           macStrs[2],
+		PortUUID:      "portuuid2",
+		OFPort:        2,
+		ContainerID:   "containerid2",
+		PodNamespace:  "namespace0",
+	},
+}
+
+var testInterfaceConfigs = []*interfacestore.InterfaceConfig{
+	{
+		InterfaceName: "interface0",
+		IP:            net.ParseIP(ipStrs[0]),
+		MAC:           macs[0],
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "portuuid0",
+			OFPort:   0,
+		},
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			ContainerID:  "containerid0",
+			PodName:      "pod0",
+			PodNamespace: "namespace0",
+		},
+	},
+	{
+		InterfaceName: "interface1",
+		IP:            net.ParseIP(ipStrs[1]),
+		MAC:           macs[1],
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "portuuid1",
+			OFPort:   1,
+		},
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			ContainerID:  "containerid1",
+			PodName:      "pod1",
+			PodNamespace: "namespace1",
+		},
+	},
+	{
+		InterfaceName: "interface2",
+		IP:            net.ParseIP(ipStrs[2]),
+		MAC:           macs[2],
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "portuuid2",
+			OFPort:   2,
+		},
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			ContainerID:  "containerid2",
+			PodName:      "pod2",
+			PodNamespace: "namespace0",
+		},
+	},
+}
+
+func parseMAC(mac string) net.HardwareAddr {
+	res, _ := net.ParseMAC(mac)
+	return res
+}
+
+func TestPodInterfaceQuery(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	testcases := map[string]struct {
+		query           string
+		expectedStatus  int
+		expectedContent Response
+	}{
+		"Hit pod interface query": {
+			query:           "?name=pod0&&namespace=namespace0&&all-namespaces=false",
+			expectedStatus:  http.StatusOK,
+			expectedContent: responses[0],
+		},
+		"Miss pod interface query": {
+			query:           "?name=pod1&&namespace=namespace0&&all-namespaces=false",
+			expectedStatus:  http.StatusNotFound,
+			expectedContent: Response{},
+		},
+	}
+
+	for k, tc := range testcases {
+		// mock GetInterfacesByType() in pkg interfacestore
```

Comment:
Comments like this are not helpful here, maybe remove them.

Reply:
Removed